src/client/Client.cc

```diff
@@ -1305,6 +1305,14 @@ Inode* Client::insert_trace(MetaRequest *request, MetaSession *session)
 
     in = add_update_inode(&ist, request->sent_stamp, session,
 			  request->perms);
+    if (in && op == CEPH_MDS_OP_GETATTR) {
+      int mask = request->head.args.getattr.mask;
+      if (mask & CEPH_STAT_RSTAT) {
+        ldout(cct, 20) << " taking inode rstat " << ist.rstat 
+          << " for " << *in << dendl;
+        in->rstat = ist.rstat;
+      }
+    }
```

Comment:
why need this. I think inode version changes when updating rstat

Reply:
Inode version is changed when updating rstat on auth MDS. When lookup request acquires rdlock on filelock on non-auth MDS first, it will sync dirstat from auth MDS along with new inode version, but no latest rstat. So client will get this new inode version but still old rstat. Later getattr request comes, it can sync latest rstat from auth MDS but still this inode version. Then client get the same inode version and won't update its rstat.

Reply:
If we make auth mds handle CEPH_STAT_RSTAT request. we can move this code into add_update_inode() (always update rstats if reply is from auth mds)